step-plans/step-plans-core/src/main/java/step/core/artefacts/reports/resolvedplan/ResolvedPlanNodeCachedAccessor.java

```diff
@@ -0,0 +1,73 @@
+package step.core.artefacts.reports.resolvedplan;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import step.core.execution.model.Execution;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Stream;
+
+public class ResolvedPlanNodeCachedAccessor  {
+
+    private static final Logger logger = LoggerFactory.getLogger(ResolvedPlanNodeCachedAccessor.class);
+    private final ResolvedPlanNodeAccessor underlyingAccessor;
+    private final Map<String, List<ResolvedPlanNode>> cacheByParentId = new HashMap<>();
```

Comment:
better safe than sorry: it's probably better to use a ConcurrentHashMap just in case.